VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -66,16 +72,46 @@ public override Task<SessionHostInfo> CreateAndStart(int instanceNumber, GameRes
                 _logger.LogException($"Failed to start process based host with instance number {instanceNumber}", exception);
                 sessionHostManager.RemoveSessionHost(sessionHostUniqueId);
                 sessionHost = null;
+                string exceptionMessage = exception.ToString();
+                CollectLogs(logFolderPathOnVm, exceptionMessage);
             }
 
             return Task.FromResult(sessionHost);
         }
 
-        public override Task CollectLogs(string processId, string logsFolder, ISessionHostManager sessionHostManager)
+        public override Task CollectLogs(string id, string logsFolder, ISessionHostManager sessionHostManager)
+        {
+            return Task.CompletedTask;
+        }
+
+        public override Task CollectLogs(string logsFolder, string exceptionMessage)
         {
             // The game server is free to read the env variable for log folder and write all output to a file in that folder.
             // If required, we can add action handlers (see SystemOperations.RunProcess for example).
             // However, keeping the file handle around can be tricky.
+            if (_systemOperations.IsOSPlatform(OSPlatform.Windows))
+            {
+                try
+                {
+                    _logger.LogVerbose("Collecting logs for failed start game process.");
+                    string destinationFileName = Path.Combine(logsFolder, ConsoleLogCaptureFileName);
+
+                    using (StreamWriter sw = File.AppendText(destinationFileName))
+                    {
+                        sw.WriteLine("{0} {1}", DateTime.Now.ToLongTimeString(), DateTime.Now.ToLongDateString());
+                        sw.WriteLine($"{exceptionMessage}");
+                    }
+                    _logger.LogVerbose($"Written logs for failed start game process to {destinationFileName}.");
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogException($"Failed to write failed start game error logs for Windows process", ex);
+                }
+                return Task.CompletedTask;
+            }
+
+            _logger.LogVerbose("Logs not written to file for non-Windows operating systems.");
```

Comment:
>Logs not written to file for non-Windows operating system [](http://example.com/codeflow?start=32&length=57)

We should confirm if we can support Linux as well.

Otherwise I would suggest we tweak the message to something like "Error logs from process execution are only available in Windows"  (or something similar)

Reply:
@dgkanatsios ?

Reply:
The issue with Linux process game servers is that they are harder to test since they are not supported on LocalMultiplayerAgent at this time.